packages/utils/src/synergy.ts

```diff
@@ -328,92 +276,192 @@ export async function getSubgroupItems(perspective, subgroupId) {
-      subgroup = subgroups[subgroups.length - 1];
-      if (newSubgroupName) subgroup.subgroupName = newSubgroupName;
-      if (newSubgroupSummary) subgroup.summary = newSubgroupSummary;
-      if (newSubgroupName || newSubgroupSummary) await subgroup.update();
-    } else {
-      // otherwise create a new subgroup
-      subgroup = new ConversationSubgroup(perspective, undefined, conversation.baseExpression);
-      subgroup.subgroupName = newSubgroupName;
-      subgroup.summary = newSubgroupSummary;
-      await subgroup.save();
+    // find the last conversation & subgroup or create new ones
+    let conversation;
+    let subgroups = [] as any;
+    let subgroupItems = [] as any[];
+    const conversations = await Conversation.query(perspective, {
+      source: channelId,
+    });
+    if (conversations.length) {
+      if (existingItem) {
+        // find the conversation & subgroup of the existing item
+        const allConversations = await Conversation.all(perspective);
+        const allSubgroups = await ConversationSubgroup.all(perspective);
+        const itemLinks = await perspective.get(
+          new LinkQuery({
+            predicate: "ad4m://has_child",
+            target: item.baseExpression,
+          })
+        );
+        const subgroup = allSubgroups.find((s) =>
+          itemLinks.find((link) => link.data.source === s.baseExpression)
+        );
+        const subgroupLinks = await perspective.get(
+          new LinkQuery({
+            predicate: "ad4m://has_child",
+            target: subgroup.baseExpression,
+          })
+        );
+        conversation = allConversations.find((c) =>
+          subgroupLinks.find((link) => link.data.source === c.baseExpression)
+        );
+        subgroups = [subgroup];
+      } else {
+        // find the latest conversation & subgroup (if last item created less than 30 mins ago)
+        const latestConversation = conversations[conversations.length - 1];
+        subgroups = await ConversationSubgroup.query(perspective, {
+          source: latestConversation.baseExpression,
+        });
+        const latestSubgroup = subgroups[subgroups.length - 1] as any;
+        const latestSubgroupItems = await getSubgroupItems(
+          perspective,
+          latestSubgroup.baseExpression
+        );
+        // calculate time since last item was created
+        const lastItem = latestSubgroupItems[latestSubgroupItems.length - 1];
+        if (lastItem) {
+          const minsSinceLastItemCreated =
+            (new Date().getTime() - new Date(lastItem.timestamp).getTime()) / (1000 * 60);
+          if (minsSinceLastItemCreated < 30) {
+            // if less than 30 mins, consider the new item part of the latest conversation
+            conversation = latestConversation;
+            subgroupItems = latestSubgroupItems;
+          }
+        }
+      }
+    }
+    if (!conversation) {
```

Comment:
I would think of this function as like a method on conversation - so maybe at least pass in the conversation as paremeter so we don't have to deal with this case here too.